niceman/distributions/base.py

```diff
@@ -40,6 +40,108 @@ def TypedList(type_):
 #
 
 class SpecObject(object):
+
+    # Empty tuple would signal that all attribs will be used while establishing
+    # the "identity" of the specobject
+    _cmp_fields = tuple()
+    # Fields of the primary interest when showing diff
+    _diff_fields = tuple()
+
+    @property
+    def _cmp_id(self):
+        if not self._cmp_fields:
+            # Might need to be gone or some custom exception
+            raise RuntimeError(
+                "Cannot establish identity of %r since _cmp_fields "
+                "are not defined", self)
```

Comment:
This message won't be formatted for you.

Reply:
Fixed.